internal/web_server/web/handlers/api/admin_api.go

```diff
@@ -449,6 +450,34 @@ func (aah *AdminApiHandler) deleteArticle(c echo.Context) error {
 	return c.NoContent(http.StatusOK)
 }
 
+// Rearrange the sequence of questions for a quiz.
+func (aah *AdminApiHandler) rearrangeQuestions(c echo.Context) error {
+	// Get the quiz ID
+	quizID, err := uuid.Parse(c.QueryParam(queryParamQuizID))
+	if err != nil {
+		return utils.Render(c, http.StatusBadRequest, components.ErrorText("error-question-list", errorInvalidQuizID))
+	}
+
+	// Get the map of question IDs and their new arrangement number.
+	// This is a map in JSON body.
+	questionsList := make(map[int]uuid.UUID)
+	err = c.Bind(&questionsList)
+	if err != nil {
+		return utils.Render(c, http.StatusBadRequest, components.ErrorText("error-question-list", "Ugyldig liste med spørsmål. Det må være en map med rekkefølge og IDer"))
+	}
+
+	// Rearrange the questions
+	err = questions.RearrangeQuestions(aah.sharedData.DB, c.Request().Context(), quizID, questionsList)
+	if err != nil {
+		if err == questions.ErrNonSequentialQuestions {
+			return utils.Render(c, http.StatusBadRequest, components.ErrorText("error-question-list", "Spørsmålene må ha en sekvensiell rekkefølge"))
+		}
+		return err
+	}
+
+	return c.NoContent(http.StatusOK)
```

Comment:
i see you're doing no content, not replacing the table. This is probably okay, unless the list of questions on the page can go out of sync with the database order?

Reply:
I send the whole list every time, so if one request in the middle isn't sent for some reason, it should update afterwards anyway. And if the last request happens to not be sent, it's not the end of the world. This is much simpler than swapping the table, which would probably add 50-100 more lines of code.